test/e2e/app_suffix_test.go

```diff
@@ -0,0 +1,179 @@
+		kapp.RunWithOpts([]string{"deploy", "-f", "-", "-a", existingName}, RunOpts{IntoNs: true, StdinReader: strings.NewReader(yaml1)})
+		NewPresentClusterResource("configmap", existingName, env.Namespace, kubectl)
+
+		os.Setenv("KAPP_EXPERIMENTAL_FQ_CONFIGMAP_NAMES", "True")
+		kapp.RunWithOpts([]string{"deploy", "-f", "-", "-a", existingName}, RunOpts{IntoNs: true, StdinReader: strings.NewReader(yaml1)})
+
+		NewMissingClusterResource(t, "configmap", existingName, env.Namespace, kubectl)
+		c := NewPresentClusterResource("configmap", existingName+app.AppSuffix, env.Namespace, kubectl)
+		require.Contains(t, c.Labels(), app.KappIsConfigmapMigratedLabelKey)
+
+		kapp.Run([]string{"delete", "-a", existingName})
+	})
+
+	// if a user has disabled the migration on an app that was already migrated
+	// the error we expect is that the app belongs to a different app
+	// - there will be a documented way to recover from this if desired
+	logger.Section("migration disabled with already migrated app", func() {
+		os.Setenv("KAPP_EXPERIMENTAL_FQ_CONFIGMAP_NAMES", "False")
+		kapp.RunWithOpts([]string{"deploy", "-f", "-", "-a", name}, RunOpts{IntoNs: true, StdinReader: strings.NewReader(yaml1)})
+
+		os.Setenv("KAPP_EXPERIMENTAL_FQ_CONFIGMAP_NAMES", "True")
+		kapp.RunWithOpts([]string{"deploy", "-f", "-", "-a", name}, RunOpts{IntoNs: true, StdinReader: strings.NewReader(yaml1)})
+
+		os.Setenv("KAPP_EXPERIMENTAL_FQ_CONFIGMAP_NAMES", "False")
+		_, err := kapp.RunWithOpts([]string{"deploy", "-f", "-", "-a", name}, RunOpts{IntoNs: true,
+			AllowError: true, StdinReader: strings.NewReader(yaml1)})
+
+		require.Containsf(t, err.Error(), "kapp: Error: Ownership errors:", "already associated with a different app")
+		// the old name configmap is created
+		NewPresentClusterResource("configmap", name, env.Namespace, kubectl)
+
+		RemoveClusterResource(t, "configmap", name, env.Namespace, kubectl)
+
+		os.Setenv("KAPP_EXPERIMENTAL_FQ_CONFIGMAP_NAMES", "True")
+		cleanUp()
+	})
+
+	os.Unsetenv("KAPP_EXPERIMENTAL_FQ_CONFIGMAP_NAMES")
+}
+
+func TestAppSuffix_ConfigmapExists_MigrationEnabled(t *testing.T) {
+	env := BuildEnv(t)
+	logger := Logger{}
+	kapp := Kapp{t, env.Namespace, env.KappBinaryPath, logger}
+	kubectl := Kubectl{t, env.Namespace, logger}
+
+	name := "test-app-suffix-configmap-exists"
+	fqName := name + app.AppSuffix
+
+	cleanUp := func() {
+		kapp.Run([]string{"delete", "-a", name})
+	}
+
+	cleanUp()
+	defer cleanUp()
+
+	os.Setenv("KAPP_EXPERIMENTAL_FQ_CONFIGMAP_NAMES", "True")
+
+	logger.Section("with suffix and not marked as a kapp-app", func() {
+		NewClusterResource(t, "configmap", fqName, env.Namespace, kubectl)
+		NewPresentClusterResource("configmap", fqName, env.Namespace, kubectl)
+
+		_, err := kapp.RunWithOpts([]string{"deploy", "-f", "-", "-a", name},
+			RunOpts{IntoNs: true, AllowError: true, StdinReader: strings.NewReader(yaml1)})
+
+		require.Containsf(t, err.Error(), "kapp: Error:", "did not contain parseable app metadata")
```

Comment:
We should first assert on err being non nil (using `require.Error()`). 
Also, we should test more than just `kapp: Error:`. Right now this will pass for all the errors.